torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -368,9 +370,34 @@ void IRPrinter::visit(const Cond* v) {
   }
 }
 
-void IRPrinter::visit(const LinearForm* v) {
-  os() << "(" << *v->getA() << ") * (" << *v->getX() << ") + (" << *v->getB()
-       << ")" << std::endl;
+void IRPrinter::visit(const Term* v) {
+  os() << "Term(";
+  v->scalar()->accept(this);
+  for (auto* t : v->variables()) {
+    os() << ",";
+    t->accept(this);
+  }
+  os() << ")";
+}
+
+void IRPrinter::visit(const Polynomial* v) {
+  bool first = true;
+  os() << "Polynomial(";
+  for (auto* t : v->variables()) {
+    emitIndent();
+    if (!first) {
+      os() << " + ";
+    }
+    first = false;
+    t->accept(this);
+    os() << "\n";
```

Comment:
This probably screws up printing of other stuff. So far the invariant has been that statements print end-of-line and handle indentation and expressions are printed in one line.

Reply:
Yes sorry this was left in debugging, will remove.